pkg/storage/ingest/reader.go

```diff
@@ -807,6 +807,10 @@ type fetchResult struct {
 	fetchedBytes int
 }
 
+func emptyFetchResult(err error) fetchResult {
```

Comment:
nit: I feel like this prefix suits this better.
```suggestion
func newEmptyFetchResult(err error) fetchResult {
```